File: spa/src/actions/dummy-command.ts
import { BaseAction } from '../action-models/base-action';

interface Input {
  id: number;
  name: string;
  scopes: string[];
}

interface ReturnType {
  message: string;
  scopes: string[];
}

export class DummyCommand extends BaseAction<Input, ReturnType> {
  getActionName(): string {
    return 'DummyCommand';
  }
}

File: DomainModules/AuctionsModule/Commands/DummyCommand.cs
using System.Collections.Generic;
using System.Linq;
using Action.Interfaces;
using Action.Models;

namespace AuctionModule.Commands
{
    public class Result
    {
        public string Message { get; set; }
        public IEnumerable<string> Scopes { get; set; }
    }

    public class DummyCommand : Command<Result>
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IEnumerable<string> Scopes { get; set; }

        protected override Result ExecuteInternal(Executor executor)
        {
            var message = $"Is it working? with name: {Name} and id: {Id}";
            var result = new Result
            {
                Message = message,
                Scopes = Scopes
            };

            return result;
        }

        public override IEnumerable<ValidationError> Validate(Executor executor)
        {
            return Enumerable.Empty<ValidationError>();
        }
    }
}
